File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def find_prime(n):
    if n < 2:
        return False
    
    for i in range(2, int(n**(1/2)) + 1):
        if n % i == 0:
            return False
        
    return True
        
def solution(numbers):
    answer = 0
    comb = []
    for i in range(1, len(numbers) + 1):
        comb += list(permutations(numbers, i))
    
    num_lst = set()
    for c in comb:
        c = int(''.join(c))
        num_lst.add(c)
    
    for n in num_lst:
        if (find_prime(n)):
            answer += 1
    
    return answer

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.js
function solution(n) {
    var answer = 0;
    
    const arr = []
    for (let i = 2; i <= n; i++) {
        arr[i] = i
    }
    
    for (let i = 2; i <= n; i++) {
        if (arr[i] === 0) continue;
        for (let j = i * 2; j <= n; j += i) {
            arr[j] = 0
        }
    }
    
    return arr.filter(e => e !== 0).length;
}